nemo/collections/llm/gpt/model/gemma2.py

```diff
@@ -264,6 +300,271 @@ def config(self) -> "Gemma2Config":
         )
 
 
+class Gemma2DotProductAttention(MegatronModule):
+    """
+    Region where selective activation recomputation is applied.
+    This region is memory intensive but less compute intensive which
+    makes activation checkpointing more efficient for LLMs (20B+).
+    See Reducing Activation Recomputation in Large Transformer Models:
+    https://arxiv.org/abs/2205.05198 for more details.
+
+    We use the following notation:
+     h: hidden size
+     n: number of attention heads
+     p: number of tensor model parallel partitions
+     b: batch size
+     s: sequence length
+    """
+
+    def __init__(
+        self,
+        config: TransformerConfig,
+        layer_number: int,
+        attn_mask_type: AttnMaskType,
+        attention_type: str,
+        attention_dropout: float = None,
+        **kwargs,
```

Comment:
where are the kwargs used?

Reply:
> where are the kwargs used?

@ananthsub not sure, but there was a fix some time ago which added kwargs so I think we want to keep them here:
https://github.com/NVIDIA/NeMo/pull/11792